### ISSUE TITLE
Enum.name reporting reportAssignmentType

### ISSUE THREAD

**Issue by ShineyDev** [microsoft/pyright#9614]
**Describe the bug**
An Enum with a member named "name" is currently invalid.

**Code or Screenshots**

```python
class Foo(enum.Enum):
  name = "NAME"  # Type "Literal['NAME']" is not assignable to declared type "property" (reportAssignmentType)
```

**VS Code extension or command-line**
pyright 1.1.391

**Comment by ShineyDev** [comment]
Oh, also: `error: "name" incorrectly overrides property of same name in class "Enum" (reportIncompatibleMethodOverride)`

**Comment by erictraut** [comment]
This is by design and not a bug. The `Enum` class defines a property `name`. You can see this in the type definition for `Enum` if you look in the `enum.pyi` typeshed stub.

**Comment by erictraut** [comment]
@ShineyDev, I consider pyright's behavior correct here. There are many things that are possible in the Python runtime that are disallowed by static type checking. The rules for static type checking are more strict on purpose. For example, assignment of an incompatible value to a typed attribute is disallowed by a static type checker but is allowed at runtime. In this case, `name` is defined as a property, so redeclaring the same symbol as some other type is an illegal redeclaration from the standpoint of a type checker.
